packages/cubejs-schema-compiler/src/compiler/CubeValidator.ts

```diff
@@ -627,18 +627,24 @@ const PolicyFilterSchema = Joi.object().keys({
     'notStartsWith',
     'endsWith',
     'notEndsWith',
+    'in',
+    'notIn',
     'gt',
     'gte',
     'lt',
     'lte',
+    'set',
+    'notSet',
     'inDateRange',
     'notInDateRange',
+    'onTheDate',
     'beforeDate',
     'beforeOrOnDate',
     'afterDate',
     'afterOrOnDate',
+    'measureFilter',
   ).required(),
-  values: Joi.func().required(),
+  values: Joi.func(),
```

Comment:
It's not great to make this optional, while most of the operators require values. So I suggest making an additional logic to mark values as optional only for set/notSet. Smth like:
```suggestion
  values: Joi.when('operator', {
    is: Joi.valid('set', 'notSet'),
    then: Joi.func().optional(),
    otherwise: Joi.func().required()
  })
```

Reply:
Oh, I didn't know joi can do this, thx :D